src/main/java/com/microsoft/graph/concurrency/ChunkedUploadProvider.java

```diff
@@ -168,14 +168,19 @@ public void upload(final List<Option> options,
         byte[] buffer = new byte[chunkSize];
 
         while (this.readSoFar < this.streamSize) {
-            int read = this.inputStream.read(buffer);
-
-            if (read == -1) {
-                break;
+            int read = 0;
+            int buffRead = 0;
+
+            while (buffRead < chunkSize) {
+                read = this.inputStream.read(buffer, buffRead, chunkSize - buffRead);
+                if (read == -1) {
+                    break;
+                }
+                buffRead += read;
             }
 
             ChunkedUploadRequest request =
-                    new ChunkedUploadRequest(this.uploadUrl, this.client, options, buffer, read,
+                    new ChunkedUploadRequest(this.uploadUrl, this.client, options, buffer, buffRead,
```

Comment:
By looking at the local scope, it seems possible to reach here with `buffRead == 0`, if we break out of the inner while loop with `read == -1`. Previous code seems to have been checking for that.

That being said, maybe previous code was over-defensive given that reaching end of stream (`read == -1`) would have also meant the `while` condition `(this.readSoFar < this.streamSize)` is false.